docs/NNNN-mutable-head.md

```diff
@@ -0,0 +1,287 @@
+  "manifest": {
+    "4d27c8...b53": [ "extensions/NNNN-mutable-head/head/content/r1/foo/bar.xml" ],
+    "9bb43j...n3a": [ "extensions/NNNN-mutable-head/head/content/r2/file1.txt" ],
+    "u8b99v...7b2": [ "extensions/NNNN-mutable-head/head/content/r3/file1.txt" ],
+    "7dcc35...c31": [ "v1/content/foo/bar.xml" ],
+    "cf83e1...a3e": [ "v1/content/empty.txt" ],
+    "ffccf6...62e": [ "v1/content/image.tiff" ]
+  },
+  "type": "https://ocfl.io/1.0/spec/#inventory",
+  "versions": {
+    "v1": {
+      "created": "2018-01-01T01:01:01Z",
+      "message": "Initial import",
+      "state": {
+        "7dcc35...c31": [ "foo/bar.xml" ],
+        "cf83e1...a3e": [ "empty.txt" ],
+        "ffccf6...62e": [ "image.tiff" ]
+      },
+      "user": {
+        "address": "alice@example.com",
+        "name": "Alice"
+      }
+    },
+    "v2": {
+      "created": "2018-02-02T02:02:02Z",
+      "message": "Fix bar.xml, remove image.tiff, add empty2.txt",
+      "state": {
+        "9bb43j...n3a": [ "file2.txt" ],
+        "u8b99v...7b2": [ "file1.txt" ],
+        "4d27c8...b53": [ "foo/bar.xml" ],
+        "cf83e1...a3e": [ "empty.txt", "empty2.txt" ]
+      },
+      "user": {
+        "address": "bob@example.com",
+        "name": "Bob"
+      }
+    }
+  }
+}
+```
+
+#### Revisions
+
+Any change that is made to a mutable HEAD is known as a revision. Each revision is assigned a revision number in the form of `rN`, where `N` is a positive integer greater than 0, similar to OCFL version numbers. When a mutable HEAD is first created, its initial revision number is `r1`. Every subsequent change that's made to the mutable HEAD MUST use the next available revision number. For example, the second change would use revision `r2` and so forth.
+
+Revisions are tracked by writing revision marker files to the revisions directory located at `[object-root]/extensions/NNNN-mutable-head/revisions`. A revision marker file MUST be named using the revision number (eg. `r1`). Revision marker files MUST only contain a single line containing the marker's revision number.
+
+A new revision marker MUST be created before applying an update to the mutable HEAD. If the revision marker already exists, this indicates that a concurrent update occurred, and the pending update MUST be aborted.
+
+#### Content Directory
+
+The mutable HEAD content directory MUST NOT contain any files that are not referenced in the mutable HEAD inventory manifest.
+
+The mutable HEAD content directory is further subdivided by directories named for revision numbers. Every file that's added to the mutable HEAD MUST be placed within the revision subdirectory that corresponds with the target revision number.
+
+For example, if `foo.txt` is added in the second revision of an object's mutable HEAD, then the files content path is `extensions/NNNN-mutable-head/head/content/r2/foo.txt`.
+
+If a revision does not add any new files, then a new content revision directory MUST NOT be created.
```

Comment:
Yes, you can update or delete files in a revision. This section was intended to specify what a content directory must look like rather than describing how various operations interact with the content directory, which should be covered in the implementation notes. Let me know if you don't think that the other operations are well enough defined and I can add additional language where appropriate.

Reply:
I think the danger of the current language is that it doesn't make is clear that a change to logical file contents will correspond to adding a file. Perhaps the language from "Mutable HEAD mutations" could be reused. Perhaps:

> If a revision does not add any files with digests not in the mutable HEAD inventory's manifest, then a new content revision directory MUST NOT be created.

Reply:
Updated

Reply:
Thanks for clarifying that